discovery-provider/src/utils/prometheus_metric.py

```diff
@@ -7,6 +8,63 @@
 logger = logging.getLogger(__name__)
 
 
+def save_duration_metric(metric_group):
+    # a decorator that takes the `metric_group` parameter to create:
+    # * a histogram for detecting duration and latency from a decorated function
+    # * a gauge for reporting the last task's duration (in seconds)
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            histogram_metric = PrometheusMetric(
```

Comment:
why have both?

Reply:
@dmanjunath Mentioned on lines 13 and 14, but mainly you can't gain raw seconds from histograms.

Here's a bit of info on [errors of quantile estimation](https://prometheus.io/docs/practices/histograms/#errors-of-quantile-estimation).